sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -40,6 +49,30 @@ uint32_t to_uint32_t(sycl::marray<bfloat16, N> x, size_t start) {
 }
 } // namespace detail
 
+// The printf() function defined lower takes arguments as is and may need
+// a conversion to some printable type for proper output.
+// The specializations of printf_arument_type_traits replaing T with some
+// printable type P must do so only when T is convertible to P.
+// Otherwise, the behavior is undefined.
+template <typename T, typename SFINAE = void>
+struct printf_arument_type_traits {
+  using type = T;
+};
+
+// simd_view to one element needs conversion to underlying element type
+// for proper output.
+template <typename T>
+struct printf_arument_type_traits<
+    T,
+    std::enable_if_t<
+        sycl::ext::intel::esimd::detail::is_one_element_simd_view<T>::value>> {
+  using type = typename T::element_type;
+};
```

Comment:
is there ever a reason we would want to print the simd_view structure for a single element instead of the value it is carrying?  is it possible to do so in some way or does this always map a single valued simd_view element to the value itself?  I can't really think of a reason we would want it to behave that way...

Reply:
@spencerpatty - excuse me, I don't understand your question and suggestion. 
This is the LIT test for the fix: https://github.com/intel/llvm-test-suite/pull/1130. Would you expect to see different output there?

Reply:
> is there ever a reason we would want to print the simd_view structure for a single element instead of the value it is carrying? 

The value carried by simd_view can be reached by explicit case to underlying type, e.g. (int)Vec[0].
Vec[0] does not point to a carried element, it only returns a simd_view structure.